src/virt.rs

```diff
@@ -562,11 +558,19 @@ impl VirtContext {
     }
 
     /// Handle the trap coming from the firmware
-    pub fn handle_firmware_trap(&mut self, mctx: &mut MiralisContext) {
+    pub fn handle_firmware_trap(&mut self, mctx: &mut MiralisContext, policy: &mut Policy) {
         let cause = self.trap_info.get_cause();
         match cause {
-            MCause::EcallFromSMode if Policy::ecall_from_firmware(self).overwrites() => {
+            MCause::EcallFromSMode if policy.ecall_from_payload(mctx, self).overwrites() => {
                 // Nothing to do, the Policy module handles those ecalls
+                log::info!("Catching E-call from payload in the policy module");
+            }
+            MCause::EcallFromUMode
+                if self.get(Register::X17) == abi::MIRALIS_EID
+                    && policy.ecall_from_firmware(mctx, self).overwrites() =>
```

Comment:
I thin the policy and the Miralis `ecall`s should be two distinct things. So each policy module can have its own `EID`.